src/main/java/schemacrawler/crawl/StreamlinerSchemaCrawler.java

```diff
@@ -193,6 +208,86 @@ public static StreamlinerCatalog getCatalog(String schemaName, String jdbcUrl,
         }
     }
 
+  private static StreamlinerCatalog getSnowflakeCatalog(
+      SnowflakeQueryHandler queryHandler,
+      String schemaName,
+      List<String> tableTypes,
+      List<String> tableWhitelist) {
+    Map<String, MutableTable> tables = new TreeMap<>();
+    Schema schema = new SchemaReference(schemaName, schemaName);
+    boolean findTables = tableTypes.contains("table");
+    boolean findViews = tableTypes.contains("view");
+
+    String snowflakeSchemaName = "";
+    List<String> tableList = new ArrayList<>();
+    {
+      for (Map<String, Object> row :
+          queryHandler.queryToList(SNOWFLAKE_TABLES_TEMPLATE, schemaName, "")) {
+        snowflakeSchemaName = (String) row.get("TABLE_SCHEMA");
+        String tableName = (String) row.get("TABLE_NAME");
+        tableList.add(String.format("'%s'", tableName));
+        if (tables.containsKey(tableName)) {
+          throw new IllegalStateException("Found table " + tableName + " twice");
+        }
+        String comment = trimIfNotNull((String) row.get("COMMENT"));
+        String tableType = (String) row.get("TABLE_TYPE");
+        if (tableType.contains("VIEW")) {
+          if (findViews) {
+            MutableView table = new MutableView(schema, tableName);
+            table.setRemarks(comment);
+            tables.put(tableName, table);
+          }
+        } else if (findTables) {
+          MutableTable table = new MutableTable(schema, tableName);
+          table.setRemarks(comment);
+          tables.put(tableName, table);
+        }
+      }
+    }
+    {
+      int snowflakeTablesCount = tableList.size();
+      for (int i = 0; i < snowflakeTablesCount; i = i + 10) {
```

Comment:
I was suspect of this logic with batches of say 11 or 21, but I think it's right.

https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html

Reply:
added a parameter "batchTableCount" in config file to change this value.